zebrad/tests/common/config.rs

```diff
@@ -53,33 +53,41 @@ pub fn default_test_config(net: &Network) -> Result<ZebradConfig> {
 
     let consensus = zebra_consensus::Config::default();
 
-    let force_use_color = !matches!(
-        env::var("ZEBRA_FORCE_USE_COLOR"),
-        Err(env::VarError::NotPresent)
-    );
+    let force_use_color = env::var("FORCE_USE_COLOR").is_ok();
 
     let mut tracing = tracing::Config::default();
     tracing.force_use_color = force_use_color;
 
     let mut state = zebra_state::Config::ephemeral();
     state.debug_validity_check_interval = Some(DATABASE_FORMAT_CHECK_INTERVAL);
 
-    // These are the ZF funding stream addresses for mainnet and testnet.
-    #[allow(unused_mut)]
+    // Provide a default miner address matched to the effective network so
+    // mining-related tests have a valid configuration. Environment variables
+    // like ZEBRA_MINING__MINER_ADDRESS will override this file value.
     let mut mining = zebra_rpc::config::mining::Config::default();
 
-    let miner_address = if network.network.is_mainnet() {
+    // Determine the effective network kind: prefer environment override if present.
+    let effective_kind: NetworkKind = match env::var("ZEBRA_NETWORK__NETWORK").ok().as_deref() {
+        Some("Mainnet") | Some("mainnet") | Some("Main") | Some("main") => NetworkKind::Mainnet,
+        Some("Regtest") | Some("regtest") => NetworkKind::Regtest,
+        Some("Testnet") | Some("testnet") => NetworkKind::Testnet,
```

Comment:
Lowercase/partial values aren't parsed as a valid `NetworkKind` (we could change this if we want, but, it's just not the case currently, and Zebra will fail to load the configuration from the environment if they're provided).

```suggestion
        Some("Mainnet") => NetworkKind::Mainnet,
        Some("Regtest") => NetworkKind::Regtest,
        Some("Testnet") => NetworkKind::Testnet,
```